Packs/Infoblox/Integrations/Infoblox/Infoblox.py

```diff
@@ -310,6 +317,116 @@ def delete_rpz_rule(self, reference_id: Optional[str]) -> Dict:
+        params = {
+            "zone": zone,
+            "_return_as_object": 1
+        }
+        res = self._http_request('GET', 'allrecords', params=params)
+        return res
+
+    def list_hosts(self):
+        params = {
+            "_return_fields": "ipv4addrs",
+            "_return_as_object": 1
+        }
+        res = self._http_request('GET', "record:host", params=params)
+        return res
+
+    def search_host_record(self, name: str):
+        params = {
+            "name": name,
+            "_return_as_object": 1
+        }
+        res = self._http_request('GET', "record:host", params=params)
+        return res
+
+    def create_record(self, suffix: Optional[str], **kwargs: Union[str, int, None]) -> Dict:
+        """Creates new record.
+        Args:
+            suffix: The infoblox object to be used as a url path.
+            kwargs: A dict of arguments to be passed to the rule body. The following may appear:
+                - name
+                - rp_zone
+                - comment
+                - ipv4addr
+                - ipv6addr
+                - mail_exchanger
+                - preference
+                - order
+                - preference
+                - replacement
+                - ptrdname
+                - priority
+                - target
+                - weight
+                - port
+                - text
+        Returns:
+            Response JSON
+        """
+        request_data = {key: val for key, val in kwargs.items() if val is not None}
+        request_params = {'_return_fields+': ','.join(request_data.keys()) + ',disable,name'}
+        rule = self._http_request('POST', suffix, data=json.dumps(request_data), params=request_params)
+        rule['result']['type'] = suffix
+        return rule
+
+    def add_host(self, host: Optional[str], ipadd: Optional[str]) -> Dict:
+        """Add a host record.
+        Args:
+            host: FQDN to change
+            ipadd: IP Address
+
+        Returns:
+            Response JSON
+        """
+        suffix = "record:host"
+        payload = '{ "name":"' + str(host) + '","ipv4addrs":[{"ipv4addr":"' + str(ipadd) + '"}]}'
+        headers = {'content-type': 'application/json', 'Accept-Charset': 'UTF-8'}
+
+        rule = self._http_request('POST', suffix, headers=headers, data=payload)
+        return rule
+
+    def update_host_ip(self, refid: str, ipv4addr: str):
+        params = {
+            "_return_fields": "ipv4addrs",
+            "_return_as_object": 1
+        }
+        payload = {
+            "ipv4addrs": [
+                {"ipv4addr": str(ipv4addr)}
+            ]
+        }
+        res = self._http_request('PUT', f"{refid}", params=params, json_data=payload)
+        return res
+
+    def update_a_record(self, refid: str, ipv4addr: str, name: Optional[str], comment: Optional[str]):
+        params = {
+            "_return_as_object": 1
+        }
+        payload = {
+            "ipv4addr": str(ipv4addr),
```

Comment:
No need for casting. It's already a string.
```suggestion
            "ipv4addr": ipv4addr,
```